src/Symfony/Component/Lock/Bridge/DynamoDb/Store/DynamoDbStore.php

```diff
@@ -0,0 +1,284 @@
+    private string $tokenAttr;
+    private string $expirationAttr;
+    private int $readCapacityUnits;
+    private int $writeCapacityUnits;
+
+    public function __construct(
+        DynamoDbClient|string $clientOrUrl,
+        array $options = [],
+        private readonly int $initialTtl = 300
+    ) {
+        if ($clientOrUrl instanceof DynamoDbClient) {
+            $this->client = $clientOrUrl;
+        } else {
+            if (!str_starts_with($clientOrUrl, 'dynamodb:')) {
+                throw new InvalidArgumentException('Unsupported DSN for DynamoDB.');
+            }
+
+            if (false === $params = parse_url($clientOrUrl)) {
+                throw new InvalidArgumentException('The given Amazon DynamoDB DSN is invalid.');
+            }
+
+            $query = [];
+            if (isset($params['query'])) {
+                parse_str($params['query'], $query);
+            }
+
+            // check for extra keys in options
+            $optionsExtraKeys = array_diff(array_keys($options), array_keys(self::DEFAULT_OPTIONS));
+            if (0 < \count($optionsExtraKeys)) {
+                throw new InvalidArgumentException(\sprintf('Unknown option found: [%s]. Allowed options are [%s].', implode(', ', $optionsExtraKeys), implode(', ', array_keys(self::DEFAULT_OPTIONS))));
+            }
+
+            // check for extra keys in query
+            $queryExtraKeys = array_diff(array_keys($query), array_keys(self::DEFAULT_OPTIONS));
+            if (0 < \count($queryExtraKeys)) {
+                throw new InvalidArgumentException(\sprintf('Unknown option found in DSN: [%s]. Allowed options are [%s].', implode(', ', $queryExtraKeys), implode(', ', array_keys(self::DEFAULT_OPTIONS))));
+            }
+
+            $options = $query + $options + self::DEFAULT_OPTIONS;
+
+            $clientConfiguration = [
+                'region' => $options['region'],
+                'accessKeyId' => rawurldecode($params['user'] ?? '') ?: $options['access_key'] ?? self::DEFAULT_OPTIONS['access_key'],
+                'accessKeySecret' => rawurldecode($params['pass'] ?? '') ?: $options['secret_key'] ?? self::DEFAULT_OPTIONS['secret_key'],
+            ];
+            if (null !== $options['session_token']) {
+                $clientConfiguration['sessionToken'] = $options['session_token'];
+            }
+            if (isset($options['debug'])) {
+                $clientConfiguration['debug'] = $options['debug'];
+            }
+            unset($query['region']);
+
+            if ('default' !== ($params['host'] ?? 'default')) {
+                $clientConfiguration['endpoint'] = \sprintf('%s://%s%s', ($options['sslmode'] ?? null) === 'disable' ? 'http' : 'https', $params['host'], ($params['port'] ?? null) ? ':'.$params['port'] : '');
+                if (preg_match(';^dynamodb\.([^\.]++)\.amazonaws\.com$;', $params['host'], $matches)) {
+                    $clientConfiguration['region'] = $matches[1];
+                }
```

Comment:
I don't see where this code is tested. Can you add a test case with a custom host?

Reply:
@GromNaN This [test](https://github.com/symfony/symfony/pull/60138/files#diff-ad209a8e7643dd458b2823bf4401a13a8b72dfb08bc09519cba84e1e2ba4cfbeR130) is setting `host => localhost` which tests that part